workers/unity/Packages/com.improbable.gdk.core/Worker/WorkerConnector.cs

```diff
@@ -196,6 +207,67 @@ protected virtual string SelectDeploymentName(DeploymentList deployments)
             return null;
         }
 
+        protected virtual string SelectLoginToken(List<LoginTokenDetails> loginTokens)
+        {
+            return null;
+        }
+
+        protected virtual string RetrievePlayerIdentityToken(string authToken, string playerId, string displayName)
+        {
+            var result = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync(
+                RuntimeConfigDefaults.LocatorHost,
+                RuntimeConfigDefaults.DevelopmentPlayerIdentityTokenPort,
+                new PlayerIdentityTokenRequest
+                {
+                    DevelopmentAuthenticationTokenId = authToken,
+                    PlayerId = playerId,
+                    DisplayName = displayName,
+                }
+            ).Get();
+
+            if (!result.HasValue)
+            {
+                throw new ConnectionFailedException("Did not receive a player identity token.",
```

Comment:
This seems like the wrong exception to throw here and elsewhere in this function and `RetrieveLoginToken`.

Reply:
would you prefer to have a new type of exception that is only related to the anonymous auth flow?

Reply:
Hmmm - either that or a new element in the `ConnectionErrorReason` like `AuthenticationFailed` or similar.

Reply:
talked offline, already had added that element in the enum, but we agree that the anon auth flow is big enough to create a new exception for this part of the connection flow. gonna change that